### DISCUSSION TITLE
What is the configuration property `packageClassList` used for?

### DISCUSSION THREAD

**Discussion by robingenz** [ionic-team/capacitor#7405] [Q&A] [answered]
There seems to be a new configuration property `packageClassList`. However, I could not find any documentation about it. What is it used for?

**Comment by jcesarmobile** [comment, marked as answer]
It's not a configuration property, it's not exposed in the definitions file.

Capacitor 6 removes the plugin auto registration as it caused crashes in very specific circumstances, and since creating a new file and adding it to a native iOS project is a bit complex, which would cause a harder migration from Capacitor 5 to Capacitor 6, it was decided to put the plugin class list in the generated `capacitor.config.json`. Capacitor 6 uses that list to register the plugins.

**Reply by robingenz** [reply]
Thanks for the insights! I was just wondering what it was needed for.

**Reply by shiv19** [reply]
@robingenz 
I was following the Plugin tutorial from the docs.
https://capacitorjs.com/docs/plugins/tutorial/ios-implementation

I noticed that the ScreenOrientationPlugin doesn't automatically get added to the packageClassList in capacitor 6.
Is there anything I should do in addition to what is mentioned in the docs to make this work?

Android side of the tutorial is working as expected.
I'll make a PR to the `start` branch of the plugin tutorial repo shortly, as I have migrated it to Cap 6 and was able to follow all the steps of the tutorial.

**Reply by shiv19** [reply]
Here is the PR with the plugin tutorial migration for the `start` branch to capacitor 6.
https://github.com/ionic-enterprise/capacitor-plugin-tutorial/pull/22#issue-2317998810

**Reply by andrewjhogue** [reply]
@shiv19 just ran into the same issue + found the solution in Capacitor 6 docs.

Seems like you have to subclass the CAPBridgeViewController class to register custom plugins:
https://capacitorjs.com/docs/ios/custom-code#register-the-plugin

**Note:** To get this to work, I also had to check **Inherit Module From Target** when changing the View Controller Class in Main.storyboard:

<img width="259" alt="Inherit Module From Target" src="https://github.com/ionic-team/capacitor/assets/169485360/c53aabb6-b681-48ff-a705-2a4c110d1b83">

**Reply by shiv19** [reply]
@andrewjhogue that must be the missing piece. Thank you 😃
